trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/system/storedprocdure/GravitinoStoredProcedureFactory.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.trino.connector.system.storedprocdure;
+
+import static com.datastrato.gravitino.trino.connector.GravitinoErrorCode.GRAVITINO_UNSUPPORTED_OPERATION;
+
+import com.datastrato.gravitino.trino.connector.catalog.CatalogConnectorManager;
+import io.trino.spi.TrinoException;
+import io.trino.spi.procedure.Procedure;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/** This class managed all the stored procedures */
+public class GravitinoStoredProcedureFactory {
+
+  private final CatalogConnectorManager catalogConnectorManager;
+  private final String metalake;
+
+  public final List<GravitinoStoredProcedure> procedures = new ArrayList<>();
```

Comment:
Why do you use `list` instead `set` here as you need to convert it to set later.

Reply:
Previously, a list was used here as well. I'm more accustomed to using lists and arrays, so we don't have to worry about errors caused by the lack of implementation of hash and equals methods.